docs/apis/subsystems/muc/index.md

```diff
@@ -0,0 +1,310 @@
+
+Once a cache object has been retrieved it operates exactly as the same as a cache that has been created for a definition.
+
+To create an ad-hoc cache you would use the following:
+
+```php
+$cache = cache::make_from_params(cache_store::MODE_APPLICATION, 'mod_myplugin', 'mycache');
+```
+
+Really don't be lazy, if you don't have a good reason to use an ad-hoc cache you should be spending a extra 5 minutes creating a definition.
+
+## The definition
+
+The above section illustrated how to create a basic definition, specifying just the area name (the key) and the mode for the definition. Those being the two required properties for a definition.<br />
+There are many other options that will let you make the most of the Cache API and will undoubtedly be required when implementing and converting cache solutions to the Cache API.
+
+The following details the options available to a definition and their defaults if not applied:
+
+```php
+$definitions = [
+    // The name of the cache area is the key. The component/plugin will be picked up from the file location.
+    'area' => [
+        'mode' => cache_store::MODE_*,
+        'simplekeys' => false,
+        'simpledata' => false,
+        'requireidentifiers' => ['ident1', 'ident2'],
+        'requiredataguarantee' => false,
+        'requiremultipleidentifiers' => false,
+        'requirelockingread' => false,
+        'requirelockingwrite' => false,
+        'requirelockingbeforewrite' => false,
+        'maxsize' => null,
+        'overrideclass' => null,
+        'overrideclassfile' => null,
+        'datasource' => null,
+        'datasourcefile' => null,
+        'staticacceleration' => false,
+        'staticaccelerationsize' => null,
+        'ttl' => 0,
+        'mappingsonly' => false,
+        'invalidationevents' => ['event1', 'event2'],
+        'canuselocalstore' => false
+        'sharingoptions' => cache_definition::SHARING_DEFAULT,
+        'defaultsharing' => cache_definition::SHARING_DEFAULT,
+    ],
+];
+```
+
+### Setting requirements
+
+The definition can specify several requirements for the cache.<br />
+This includes identifiers that must be provided when creating the cache object, that the store guarantees data stored in it will remain there until removed, a store that supports multiple identifiers, and finally read/write locking.
+The options for these are as follows:
+; simplekeys : [](bool) Set to true if your cache will only use simple keys for its items.<br />Simple keys consist of digits, underscores and the 26 chars of the english language. a-zA-Z0-9_<br />If true the keys won't be hashed before being passed to the cache store for gets/sets/deletes. It will be better for performance and possible only because we know the keys are safe.
+; simpledata : [](bool) If set to true we know that the data is scalar or array of scalar.<br />If true, the data values will be stored as they are. Otherwise they will be serialised first.
```

Comment:
```suggestion
- **simpledata**. [bool] If set to true we know that the data is scalar or array of scalar.<br />If true, the data values will be stored as they are. Otherwise, they will be serialised first.
```